nano/node/node.cpp

```diff
@@ -977,6 +1004,125 @@ void nano::node::ongoing_unchecked_cleanup ()
 	});
 }
 
+bool nano::node::collect_ledger_pruning_targets (std::deque<nano::block_hash> & pruning_targets_a, nano::account & last_account_a, uint64_t const batch_read_size_a, uint64_t const max_depth_a, uint64_t const cutoff_time_a)
```

Comment:
This can now be written with the parallelized table loop, but maybe not worth it if the time to read the table is much less than the time to delete.

Reply:
I think with frequent pruning deleting can take much less time while search would be long